pkgs/development/python-modules/pysvn/default.nix

```diff
@@ -8,52 +9,70 @@
 
 buildPythonPackage rec {
   pname = "pysvn";
-  version = "1.8.0";
-  disabled = isPy3k;
+  version = "1.9.12";
   format = "other";
 
   src = fetchurl {
     url = "http://pysvn.barrys-emacs.org/source_kits/${pname}-${version}.tar.gz";
-    sha256 = "0srjr2qgxfs69p65d9vvdib2lc142x10w8afbbdrqs7dhi46yn9r";
+    sha256 = "sRPa4wNyjDmGdF1gTOgLS0pnrdyZwkkH4/9UCdh/R9Q=";
   };
 
-  buildInputs = [ pkgs.subversion pkgs.apr pkgs.aprutil pkgs.expat pkgs.neon pkgs.openssl ]
-    ++ (if stdenv.isLinux then [pkgs.e2fsprogs] else []);
+  buildInputs = with pkgs; [ bash subversion apr aprutil expat neon openssl ]
+    ++ (if stdenv.isLinux then [ pkgs.e2fsprogs ] else []);
 
   # There seems to be no way to pass that path to configure.
   NIX_CFLAGS_COMPILE="-I${pkgs.aprutil.dev}/include/apr-1";
 
+  prePatch = ''
+    sed -i "117s|append(|insert(0, |" Tests/benchmark_diff.py
+  '';
+
   preConfigure = ''
     cd Source
     ${python.interpreter} setup.py backport
     ${python.interpreter} setup.py configure \
       --apr-inc-dir=${pkgs.apr.dev}/include \
       --apu-inc-dir=${pkgs.aprutil.dev}/include \
+      --pycxx-dir=${pythonPackages.pycxx.dev}/include \
+      --svn-inc-dir=${pkgs.subversion.dev}/include/subversion-1 \
+      --pycxx-src-dir=${pythonPackages.pycxx.dev}/src \
       --apr-lib-dir=${pkgs.apr.out}/lib \
       --svn-lib-dir=${pkgs.subversion.out}/lib \
-      --svn-bin-dir=${pkgs.subversion.out}/bin \
-      --svn-root-dir=${pkgs.subversion.dev}
+      --svn-bin-dir=${pkgs.subversion.out}/bin
   '' + (if !stdenv.isDarwin then "" else ''
     sed -i -e 's|libpython2.7.dylib|lib/libpython2.7.dylib|' Makefile
   '');
 
-  checkPhase = "make -C ../Tests";
+  checkInputs = [ pkgs.glibcLocales  ];
+  checkPhase = ''
+    runHook preCheck
+
+    # It is not only shebangs, some tests also write scripts dynamically
+    # so it is easier to simply search and replace
+    sed -i "s|/bin/bash|${pkgs.bash}/bin/bash|" ../Tests/test-*.sh
```

Comment:
We can't use patchShebangs here?

Reply:
no test 04 is writing its own script for the tests so we would still need this additionnal sed, so no real interest in using patchShebangs